mslib/mswms/gallery_builder.py

```diff
@@ -712,7 +712,7 @@ def add_image(path, plot, plot_object, generate_code=False, sphinx=False, url_pr
             markdown = add_times(itime, [vtime], markdown)
         plot_htmls[f"{l_type}_{dataset}{plot_object.name}"] = markdown
 
-    id = img_path.split(f"-{level}")[0]
+    id = img_path.split("-" + f"{level}".replace(" ", "_").replace(":", "_").replace("-", "_"))[0]
```

Comment:
That solves a different problem.
But personally, the mechanism of trying to "guess" poorly thought out filenames in the JavaScript is the root problem.
It would be much, much safer to build an associative array while creating the plots and store it in a static structure.

*Or* add a div element for all plots and simply toggle the visibility.  I am not enough of a web programmer to see, if this'd be a good idea.
The current approach seems efficient, but dangerous.